command/agent/agent_endpoint.go

```diff
@@ -176,3 +176,90 @@ func (s *HTTPServer) AgentDeregisterService(resp http.ResponseWriter, req *http.
 	serviceID := strings.TrimPrefix(req.URL.Path, "/v1/agent/service/deregister/")
 	return nil, s.agent.RemoveService(serviceID, true)
 }
+
+func (s *HTTPServer) AgentServiceMaintenance(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	// Only PUT supported
+	if req.Method != "PUT" {
```

Comment:
Sooooo if we want to be pedantic, the API method here should technically be [PUT](http://restcookbook.com/HTTP%20Methods/patch/), not PUT. PATCH is used to update a partial resource, but PUT is used to update the entire resource.